pkg/auth/iam/iam.go

```diff
@@ -208,6 +208,90 @@ func (filter *Filter) Auth(opts ...FilterOption) restful.FilterFunction {
 	}
 }
 
+// AuthWithoutSubdomainValidation returns a filter that filters request with valid access token in auth header or cookie
+// The difference with Auth() is this function won't check for subdomain validation even if it's active in configuration.
+//
+// The token's claims will be passed in the request.attributes["JWTClaims"] = *iam.JWTClaims{}
+// This filter is expandable through FilterOption parameter
+// Example:
+// iam.AuthWithoutSubdomainValidation(
+//
+//	WithValidUser(),
+//	WithPermission("ADMIN"),
+//
+// )
+func (filter *Filter) AuthWithoutSubdomainValidation(opts ...FilterOption) restful.FilterFunction {
```

Comment:
when should we use this function instead of **Auth()**?

Reply:
When the endpoints need to be called from page with no subdomain. For example, all endpoints that needed to be called from self-service checkout page

Reply:
I have no concerns about the design, but the implementation is not good with so many duplicates

Reply:
Thanks for the suggestion. I've moved it to a function and add parameter to it to remove the duplication